cargo build error

## Required information

**Operating system:**
E.g. Ubuntu 20.04 LTS

**Rust version:**
E.g. Rust 1.55.0

**Observed result or behaviour:**
A clear and precise description of the observed result.

**Expected result or behaviour:**
What do you expect to happen?

**Conditions where it occurred / Performed steps:**
Describe how one can reproduce the bug.


`root@MSI:~/2.iceoryx-rs/iceoryx-rs# cargo build
   Compiling iceoryx-rs v0.0.10 (/root/2.iceoryx-rs/iceoryx-rs)
The following warnings were emitted during compilation:

warning: src/runtime.rs:8:14: fatal error: iceoryx_posh/runtime/posh_runtime.hpp: No such file or directory
warning:     8 |     #include "iceoryx_posh/runtime/posh_runtime.hpp"
warning:       |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: compilation terminated.

error: failed to run custom build command for `iceoryx-rs v0.0.10 (/root/2.iceoryx-rs/iceoryx-rs)`

Caused by:
  process didn't exit successfully: `/root/2.iceoryx-rs/iceoryx-rs/target/debug/build/iceoryx-rs-3895b6884be6cb5f/build-script-build` (exit status: 1)
  --- stdout
  Output { status: ExitStatus(unix_wait_status(0)), stdout: "", stderr: "HEAD is now at ebdfcc5df Merge pull request #1312 from ApexAI/iox-#1311-increase-version-to-2.0.1\n" }
  Output { status: ExitStatus(unix_wait_status(0)), stdout: "", stderr: "" }
  Output { status: ExitStatus(unix_wait_status(0)), stdout: "-- Configuring done\n-- Generating done\n-- Build files have been written to: /root/2.iceoryx-rs/iceoryx-rs/target/iceoryx-build/iceoryx_hoofs\n", stderr: "" }
  Output { status: ExitStatus(unix_wait_status(512)), stdout: "[ 17%] Built target iceoryx_platform\n[ 20%] Building CXX object CMakeFiles/iceoryx_hoofs.dir/source/posix_wrapper/access_control.cpp.o\n", stderr: "In file included from /root/2.iceoryx-rs/iceoryx-rs/target/iceoryx-git/iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/access_control.hpp:23,\n                 from /root/2.iceoryx-rs/iceoryx-rs/target/iceoryx-git/iceoryx_hoofs/source/posix_wrapper/access_control.cpp:18:\n/root/2.iceoryx-rs/iceoryx-rs/target/iceoryx-git/iceoryx_hoofs/platform/linux/include/iceoryx_hoofs/platform/acl.hpp:20:10: fatal error: sys/acl.h: No such file or directory\n   20 | #include <sys/acl.h>\n      |          ^~~~~~~~~~~\ncompilation terminated.\nmake[2]: *** [CMakeFiles/iceoryx_hoofs.dir/build.make:271: CMakeFiles/iceoryx_hoofs.dir/source/posix_wrapper/access_control.cpp.o] Error 1\nmake[1]: *** [CMakeFiles/Makefile2:96: CMakeFiles/iceoryx_hoofs.dir/all] Error 2\nmake: *** [Makefile:130: all] Error 2\n" }
  Output { status: ExitStatus(unix_wait_status(0)), stdout: "", stderr: "" }
  Output { status: ExitStatus(unix_wait_status(256)), stdout: "-- Configuring incomplete, errors occurred!\nSee also \"/root/2.iceoryx-rs/iceoryx-rs/target/iceoryx-build/iceoryx_posh/CMakeFiles/CMakeOutput.log\".\nSee also \"/root/2.iceoryx-rs/iceoryx-rs/target/iceoryx-build/iceoryx_posh/CMakeFiles/CMakeError.log\".\n", stderr: "CMake Error at CMakeLists.txt:24 (find_package):\n  By not providing \"Findiceoryx_hoofs.cmake\" in CMAKE_MODULE_PATH this\n  project has asked CMake to find a package configuration file provided by\n  \"iceoryx_hoofs\", but CMake did not find one.\n\n  Could not find a package configuration file provided by \"iceoryx_hoofs\"\n  with any of the following names:\n\n    iceoryx_hoofsConfig.cmake\n    iceoryx_hoofs-config.cmake\n\n  Add the installation prefix of \"iceoryx_hoofs\" to CMAKE_PREFIX_PATH or set\n  \"iceoryx_hoofs_DIR\" to a directory containing one of the above files.  If\n  \"iceoryx_hoofs\" provides a separate development package or SDK, be sure it\n  has been installed.\n\n\n" }
  Output { status: ExitStatus(unix_wait_status(512)), stdout: "", stderr: "make: *** No rule to make target 'install'.  Stop.\n" }
  TARGET = Some("x86_64-unknown-linux-gnu")
  OPT_LEVEL = Some("0")
  HOST = Some("x86_64-unknown-linux-gnu")
  CXX_x86_64-unknown-linux-gnu = None
  CXX_x86_64_unknown_linux_gnu = None
  HOST_CXX = None
  CXX = None
  CXXFLAGS_x86_64-unknown-linux-gnu = None
  CXXFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CXXFLAGS = None
  CXXFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-I" "/root/2.iceoryx-rs/iceoryx-rs" "-I" "/root/2.iceoryx-rs/iceoryx-rs/target/iceoryx-install/include/iceoryx/v2.0.1" "-Wall" "-Wextra" "-Wno-noexcept-type" "-std=c++14" "-o" "/root/2.iceoryx-rs/iceoryx-rs/target/debug/build/iceoryx-rs-e6a6ce3817c727f2/out/rust_cpp/cpp_closures.o" "-c" "/root/2.iceoryx-rs/iceoryx-rs/target/debug/build/iceoryx-rs-e6a6ce3817c727f2/out/rust_cpp/cpp_closures.cpp"
  cargo:warning=src/runtime.rs:8:14: fatal error: iceoryx_posh/runtime/posh_runtime.hpp: No such file or directory
  cargo:warning=    8 |     #include "iceoryx_posh/runtime/posh_runtime.hpp"
  cargo:warning=      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=compilation terminated.
  exit status: 1

  --- stderr


  error occurred: ToolExecError: Command "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-I" "/root/2.iceoryx-rs/iceoryx-rs" "-I" "/root/2.iceoryx-rs/iceoryx-rs/target/iceoryx-install/include/iceoryx/v2.0.1" "-Wall" "-Wextra" "-Wno-noexcept-type" "-std=c++14" "-o" "/root/2.iceoryx-rs/iceoryx-rs/target/debug/build/iceoryx-rs-e6a6ce3817c727f2/out/rust_cpp/cpp_closures.o" "-c" "/root/2.iceoryx-rs/iceoryx-rs/target/debug/build/iceoryx-rs-e6a6ce3817c727f2/out/rust_cpp/cpp_closures.cpp" with args "c++" did not execute successfully (status code exit status: 1).`

--- a/build.rs
+++ b/build.rs
@@ -10,6 +10,7 @@ use std::process::Command;
 
 fn make_and_install(source_dir: &str, build_dir: &str, install_dir: &str) -> std::io::Result<()> {
     let cmake_install_prefix = format!("-DCMAKE_INSTALL_PREFIX={}", install_dir);
+    let cmake_prefix_path = format!("-DCMAKE_PREFIX_PATH={}", install_dir);
 
     for iceoryx_component in &["iceoryx_hoofs", "iceoryx_posh"] {
         let component_source_dir = format!("{}/{}", source_dir, iceoryx_component);
@@ -31,6 +32,7 @@ fn make_and_install(source_dir: &str, build_dir: &str, install_dir: &str) -> std
             .args(&[
                 "-DCMAKE_BUILD_TYPE=Release",
                 "-DBUILD_SHARED_LIBS=OFF",
+                &cmake_prefix_path,
                 &cmake_install_prefix,
                 &component_source_dir,
             ])

--- a/build.rs
+++ b/build.rs
@@ -1,8 +1,10 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
 // SPDX-FileContributor: Mathias Kraus
+// SPDX-FileContributor: Apex.AI
 
 use std::env;
+use std::io::{Error, ErrorKind};
 use std::path::Path;
 use std::process::Command;
 
@@ -13,16 +15,18 @@ fn make_and_install(source_dir: &str, build_dir: &str, install_dir: &str) -> std
         let component_source_dir = format!("{}/{}", source_dir, iceoryx_component);
         let component_build_dir = format!("{}/{}", build_dir, iceoryx_component);
 
-        Command::new("mkdir")
+        if !Command::new("mkdir")
             .args(&["-p", &component_build_dir])
-            .output()
-            .map_err(|out| {
-                println!("{:?}", out);
-                out
-            })
-            .map(|out| println!("{:?}", out))?;
+            .status()?
+            .success()
+        {
+            return Err(Error::new(
+                ErrorKind::Other,
+                format!("Could not create build dir for '{}'!", iceoryx_component),
+            ));
+        }
 
-        Command::new("cmake")
+        if !Command::new("cmake")
             .current_dir(&component_build_dir)
             .args(&[
                 "-DCMAKE_BUILD_TYPE=Release",
@@ -30,22 +34,26 @@ fn make_and_install(source_dir: &str, build_dir: &str, install_dir: &str) -> std
                 &cmake_install_prefix,
                 &component_source_dir,
             ])
-            .output()
-            .map_err(|out| {
-                println!("{:?}", out);
-                out
-            })
-            .map(|out| println!("{:?}", out))?;
+            .status()?
+            .success()
+        {
+            return Err(Error::new(
+                ErrorKind::Other,
+                format!("Could not run cmake for '{}'!", iceoryx_component),
+            ));
+        }
 
-        Command::new("cmake")
+        if !Command::new("cmake")
             .current_dir(&component_build_dir)
             .args(&["--build", ".", "--target", "install"])
-            .output()
-            .map_err(|out| {
-                println!("{:?}", out);
-                out
-            })
-            .map(|out| println!("{:?}", out))?;
+            .status()?
+            .success()
+        {
+            return Err(Error::new(
+                ErrorKind::Other,
+                format!("Could not build '{}'!", iceoryx_component),
+            ));
+        }
     }
 
     Ok(())
@@ -90,7 +98,7 @@ fn main() -> std::io::Result<()> {
     let iceoryx_build_dir = format!("{}/{}/{}", current_dir, "target", "iceoryx-build");
     let iceoryx_install_dir = format!("{}/{}/{}", current_dir, "target", "iceoryx-install");
 
-    const ICEORYX_VERSION: &str = "v2.0.1";
+    const ICEORYX_VERSION: &str = "v2.0.2";
     const ICEORYX_GIT_BRANCH: &str = ICEORYX_VERSION;
     clone_repo(
         "https://github.com/eclipse-iceoryx/iceoryx.git",

--- a/src/sb/ffi.rs
+++ b/src/sb/ffi.rs
@@ -267,7 +267,9 @@ impl Subscriber {
 
 impl fmt::Debug for Subscriber {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{:#?}", self._opaque)
+        f.debug_struct("Subscriber")
+            .field("_opaque", &self._opaque)
+            .finish()
     }
 }
 

--- a/src/sb/ffi.rs
+++ b/src/sb/ffi.rs
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
 // SPDX-FileContributor: Mathias Kraus
+// SPDX-FileContributor: Apex.AI
 
 use crate::sb::SubscriberOptions;
 
@@ -266,7 +267,7 @@ impl Subscriber {
 
 impl fmt::Debug for Subscriber {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{:#?}", self)
+        write!(f, "{:#?}", self._opaque)
     }
 }
 

